### PR TITLE
[SenecHome] Add writeable Charging Modes

### PR DESCRIPTION
**Pull request by BigFood2307**
This PR improves the SenecHome Binding by adding access to two writable charging modes for the battery systems.
This means adding the general capability of write commands to the binding and implementing the HTTPS commands for the necessary parameters.
It was shortly discussed in the [Bindings Forum thread](https://community.openhab.org/t/senechome-binding/108152/40?u=bigfood2307)
With this addition the user can now force the battery system to charge to full or enter it into storage/transport mode.

Only one other person was able to test the new version. They had issues, although that might have been due to missmatched openHab versions.
On my own system, the updated binding has been running for ~6 months without issue.
[Here](https://github.com/BigFood2307/openhab-addons/releases/tag/senechome-charge-modes-1) is the binary I have been using for oh 4.2

Direct link to jfrog for convenience, since the template asked me to :)
https://openhab.jfrog.io/ui/native/libs-pullrequest-local/org/openhab/addons/bundles/org.openhab.binding.senechome/

### CODE REVIEW

**Comment by BigFood2307** [comment]
> How do these two charging modes relate? From the descriptions they seem conflicting, If you turn on both switches, what will happen?

True, they are conflicting. I think I tested it some time ago and turning a second mode on, it would turn the first one back off.
I can test that again on the weekend.


> Did you consider to have one `chargeMode` channel, that contains a list of possible options like: `lithium charge`, `safe charge`, `nomale charge` mode etc. ?

A single channel with 3 options never entered my mind, but could actually be a good idea. I will look into how I can implement fixed state options for the channel.

**Comment by BigFood2307** [comment]
Thanks for the feedback!
I did some testing and can now confirm, that Storage Mode always just overrides Safe Charge Mode.
I still think the idea of combining them into one channel was good, so I implemented that now.

File: bundles/org.openhab.binding.senechome/README.md [hunk 1]
```diff
@@ -57,6 +60,8 @@ The property `limitationTresholdValue` is used as threshold for channel `powerLi
 | batteryFuelCharge             | percent        | Fuel charge of your battery (0 - 100%)                                   |
 | systemState                   |                | Text describing current action of the senec home system (e.g. CHARGE)    |
 | systemStateValue              |                | Value describing current action of the senec home system (e.g. 14)       |
+| safeChargeMode                | OFF/CHARGE/    | In safeChargeMode, the battery will try to fill as quickly as possible   |
+|                               | STORAGE        | in liStorageMode, the battery will try to reach 25% SOC                  |
```
Comment (lsiepel) [inline bundles/org.openhab.binding.senechome/README.md:64]:
```suggestion
| chargeMode                | OFF/CHARGE/    | In `CHARGE` mode, the battery will try to fill as quickly as possible   |
|                               | STORAGE        | in `STORAGE` mode, the battery will try to reach 25% SOC                  |
```

You might need to adjust the whirespaces,

File: bundles/org.openhab.binding.senechome/README.md [hunk 2]
```diff
@@ -19,6 +21,7 @@ Examples: Lights, pool filters, wash machines, ...
 - not equipped battery packs will return 0 for all ...Pack channels
 - currently channels for the first wallbox are implemented (senec could handle 4 wallboxes)
 - Senec disables http access at ~30.08.2023
+- Lithium Storage Mode is intended (according to the manual) for disassembly and transport. It is untested if it has any side effects.
```
Comment (lsiepel) [inline bundles/org.openhab.binding.senechome/README.md:24]:
```suggestion
- The chargeMode `STORAGE` also known as Lithium Storage Mode is intended (according to the manual) for disassembly and transport. It is untested if it has any side effects.
```

File: bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeApi.java [hunk 3]
```diff
@@ -72,6 +73,39 @@ public void setHostname(String hostname) {
      */
     public SenecHomeResponse getStatistics()
             throws TimeoutException, ExecutionException, IOException, InterruptedException, JsonSyntaxException {
+
+        String dataToSend = gson.toJson(new SenecHomeResponse());
+        ContentResponse response = postRequest(dataToSend);
+        String responseString = response.getContentAsString();
+        return Objects.requireNonNull(gson.fromJson(responseString, SenecHomeResponse.class));
```
Comment (lsiepel) [inline bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeApi.java:80]:
```suggestion
        return Objects.requireNonNull(gson.fromJson(response.getContentAsString(), SenecHomeResponse.class));
```

File: bundles/org.openhab.binding.senechome/src/main/resources/OH-INF/i18n/senechome.properties [hunk 4]
```diff
@@ -84,6 +85,7 @@ channel-type.senechome.powerLimitationState.label = Limitation State
 channel-type.senechome.powerMpp1.label = Power MPP1
 channel-type.senechome.powerMpp2.label = Power MPP2
 channel-type.senechome.powerMpp3.label = Power MPP3
+channel-type.senechome.saveChargeMode.label = Save Charge Mode
```
Comment (lsiepel) [inline bundles/org.openhab.binding.senechome/src/main/resources/OH-INF/i18n/senechome.properties:88]: These are old valuesw, don't forget to run the i18n tool: `mvn i18n:generate-default-translations`

File: bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java [hunk 5]
```diff
@@ -314,14 +343,38 @@ protected void updateStringStateFromInt(String channelName, String senecValue,
...
+            } else {
+                updateState(channel.getUID(), new StringType(STATE_SENEC_CHARGE_MODE_OFF));
+            }
+        }
+    }
+
     protected void updateDecimalState(String channelName, String senecValue) {
         Channel channel = getThing().getChannel(channelName);
         if (channel != null) {
+
```
Comment (lsiepel) [inline bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java:364]:
```suggestion
```

File: bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeBindingConstants.java [hunk 6]
```diff
@@ -20,6 +20,7 @@
  * used across the whole binding.
  *
  * @author Steven Schwarznau - Initial contribution
+ * @author Lukas Pindl - Update for writing to safeChargeMode
```
Comment (lsiepel) [inline bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeBindingConstants.java:23]:
```suggestion
 * @author Lukas Pindl - Update for writing to chargeMode
```

File: bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java [hunk 7]
```diff
@@ -58,6 +58,7 @@
  *
  * @author Steven Schwarznau - Initial contribution
  * @author Erwin Guib - added more channels, added some convenience methods to reduce code duplication
+ * @author Lukas Pindl - Update for writing to safeChargeMode
```
Comment (lsiepel) [inline bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java:61]:
```suggestion
 * @author Lukas Pindl - Update for writing to chargeMode
```

**Review by lsiepel — CHANGES_REQUESTED** [review]
Some textual improvements now this has changed. I think this was a good move to this list instead of the switches, thank you. We expect a milestone this weekend, so if you are able to look at it today we can get it into 4.3.0m2

**Comment by BigFood2307** [comment]
Should now be fixed 👍

**Review by lsiepel — APPROVED** [review]
Thanks LGTM

**Pull request merged**